workbench/src/main/setupAddRemovePlugin.js

```diff
@@ -11,59 +11,112 @@ import { settingsStore } from './settingsStore';
 
 const logger = getLogger(__filename.split('/').slice(-1)[0]);
 
+/**
+ * Spawn a child process and log its stdout, stderr, and any error in spawning.
+ *
+ * child_process.spawn is called with the provided cmd, args, and options,
+ * and the windowsHide option set to true.
+ *
+ * Required properties missing from the store are initialized with defaults.
+ * Invalid properties are reset to defaults.
+ * @param  {string} cmd - command to pass to spawn
+ * @param  {Array} args - command arguments to pass to spawn
+ * @param  {object} options - options to pass to spawn.
+ * @returns {Promise} resolves when the command finishes with exit code 0.
+ *                    Rejects with error otherwise.
+ */
+function spawnWithLogging(cmd, args, options) {
+  logger.info(cmd, args);
+  const cmdProcess = spawn(cmd, args, { ...options, windowsHide: true });
+  if (cmdProcess.stdout) {
+    cmdProcess.stderr.on('data', (data) => logger.info(data.toString()));
+    cmdProcess.stdout.on('data', (data) => logger.info(data.toString()));
+  }
+  return new Promise((resolve, reject) => {
+    cmdProcess.on('error', (err) => {
+      logger.error(err);
+      reject(err);
+    });
+    cmdProcess.on('close', (code) => {
+      if (code === 0) {
+        resolve(code);
+      } else {
+        reject(code);
+      }
+    });
+  });
+}
+
 export function setupAddPlugin() {
   ipcMain.handle(
     ipcMainChannels.ADD_PLUGIN,
-    (e, pluginURL) => {
+    async (e, pluginURL) => {
       logger.info('adding plugin at', pluginURL);
 
+      const mamba = settingsStore.get('mamba');
+      let envName;
+      let pluginID;
+      let pluginName;
+      let pluginPyName;
+
       try {
         // Create a temporary directory and check out the plugin's pyproject.toml
         const tmpPluginDir = fs.mkdtempSync(upath.join(tmpdir(), 'natcap-invest-'));
-        execSync(
-          `git clone --depth 1 --no-checkout ${pluginURL} "${tmpPluginDir}"`,
-          { stdio: 'inherit', windowsHide: true }
-        );
-        execSync('git checkout HEAD pyproject.toml', { cwd: tmpPluginDir, stdio: 'inherit', windowsHide: true });
+        await spawnWithLogging(
+          'git',
+          ['clone', '--depth', '1', '--no-checkout', pluginURL, tmpPluginDir]
+        ).then(async () => {
```

Comment:
@emlys , forgive me for popping in here. I don't think we need the `.then` chaining now that we're using `await` and resolving the promise when the spawn process closes. I think you should just be able to do like this,

```javascript
await spawnWithLoggin('git clone ...');
await spawnWithLogging('git checkout ...');
...
```